test/Microsoft.AspNet.Mvc.Core.Test/ModelBinding/Metadata/ModelMetadataProviderTest.cs

```diff
@@ -421,6 +423,60 @@ public void DisplayAttribute_Description()
         }
 
         [Fact]
+        public void DisplayName_FromResources_GetsRecomputed()
+        {
+            // Arrange
+            var display = new DisplayAttribute()
+            {
+                Name = nameof(TestResources.DisplayAttribute_Name2),
+                ResourceType = typeof(TestResources),
+            };
+
+            var provider = CreateProvider(new[] { display });
+            var metadata = provider.GetMetadataForType(typeof(string));
+            var currentCulture = Thread.CurrentThread.CurrentUICulture;
+
+            // Act & Assert
+            var result = metadata.DisplayName;
+            Assert.Equal("name from resources" + currentCulture.Name, result);
+
+            var cultureString = "fr-FR";
+            using (new CultureReplacer(uiCulture: cultureString))
+            {
+                // Reading it again ensures that we recompute.
```

Comment:
With change above, update this comment to `// Later iteration ensures value is recomputed` or (if you decide on 3 UI cultures) `// Later iterations ensure value is recomputed.`